core/domain/stats_domain.py

```diff
@@ -1351,3 +1365,360 @@ def validate(self):
+                            self.state_reference))
+            else:
+                raise utils.ValidationError(
+                    'The entity type should be either exploration or question, '
+                    'but received %s' % (self.entity_type))
+
+        if not isinstance(self.interaction_id, basestring):
+            raise utils.ValidationError(
+                'Expected interaction_id to be a string, received %s' % str(
+                    self.interaction_id))
+
+        if self.interaction_id in feconf.INTERACTION_IDS_WITHOUT_ANSWER_DETAILS:
+            raise utils.ValidationError(
+                'The %s interaction does not support soliciting '
+                'answer details from learners.' % (self.interaction_id))
+
+        if not isinstance(self.learner_answer_info_list, list):
+            raise utils.ValidationError(
+                'Expected learner_answer_info_list to be a list, '
+                'received %s' % str(self.learner_answer_info_list))
+
+        for learner_answer_info in self.learner_answer_info_list:
+            learner_answer_info.validate()
+
+        if not isinstance(self.schema_version, int):
+            raise utils.ValidationError(
+                'Expected schema_version to be an int, '
+                'received %s' % self.schema_version)
+
+        if not isinstance(self.accumulated_answer_info_json_size_bytes, int):
+            raise utils.ValidationError(
+                'Expected accumulated_answer_info_json_size_bytes to be an int '
+                'received %s' % self.accumulated_answer_info_json_size_bytes)
+
+    def add_learner_answer_info(self, learner_answer_info):
+        """Adds new learner answer info in the learner_answer_info_list.
+
+        Args:
+            learner_answer_info: LearnerAnswerInfo. The learner answer info
+                object, which is created after the learner has sbumitted the
+                details of the answer.
+        """
+        learner_answer_info_dict_size = (
+            learner_answer_info.get_learner_answer_info_dict_size())
+        if (self.accumulated_answer_info_json_size_bytes +
+                learner_answer_info_dict_size <= (
+                    MAX_LEARNER_ANSWER_INFO_LIST_BYTE_SIZE)):
+            self.learner_answer_info_list.append(learner_answer_info)
+            self.accumulated_answer_info_json_size_bytes += (
+                learner_answer_info_dict_size)
+
+    def delete_learner_answer_info(self, learner_answer_info_id):
+        """Delete the learner answer info from the learner_answer_info_list.
+
+        Args:
+            learner_answer_info_id: str. The learner answer info
+                id, which needs to be deleted from
+                the learner_answer_info_list.
+        """
+        deleted_learner_answer_info_id = (
+            learner_answer_info_id)
+        new_learner_answer_info_list = []
+        for learner_answer_info in self.learner_answer_info_list:
```

Comment:
Maybe raise an error if such an object is not found?

Reply:
Done